source/nodejs/adaptivecards/src/carousel.ts

```diff
@@ -0,0 +1,359 @@
+
+    getItemCount(): number {
+        return this._pages.length;
+    }
+
+    getItemAt(index: number): CardElement {
+        return this._pages[index];
+    }
+
+    removeItem(item: CardElement): boolean {
+        if (item instanceof CarouselPage) {
+            const itemIndex = this._pages.indexOf(item);
+
+            if (itemIndex >= 0) {
+                this._pages.splice(itemIndex, 1);
+
+                item.setParent(undefined);
+
+                this.updateLayout();
+
+                return true;
+            }
+        }
+
+        return false;
+    }
+
+    getFirstVisibleRenderedItem(): CardElement | undefined {
+        if (this.renderedElement && this._renderedPages?.length > 0) {
+            return this._renderedPages[0];
+        }
+        else {
+            return undefined;
+        }
+    }
+
+    getLastVisibleRenderedItem(): CardElement | undefined {
+        if (this.renderedElement && this._renderedPages?.length > 0) {
+            return this._renderedPages[this._renderedPages.length - 1];
+        }
+        else {
+            return undefined;
+        }
+    }
+
+    get currentPageId(): string | undefined {
+        if (this._swiper?.slides?.length) {
+            const activeSlide = this._swiper.slides[this._swiper.activeIndex] as HTMLElement;
+            return activeSlide.id;
+        }
+        return undefined;
+    }
+
+    protected internalParse(source: any, context: SerializationContext) {
+        super.internalParse(source, context);
+
+        this._pages = [];
+        this._renderedPages = [];
+
+        const jsonPages = source["pages"];
+        if (Array.isArray(jsonPages)) {
+            for (const item of jsonPages) {
+                const page = this.createCarouselPageInstance(item, context);
+                if (page) {
+                    this._pages.push(page);
+                }
+            }
+        }
+    }
+
+    protected internalToJSON(target: PropertyBag, context: SerializationContext) {
+        super.internalToJSON(target, context);
+
+        context.serializeArray(target, "pages", this._pages);
+    }
+
+    protected internalRender(): HTMLElement | undefined {
+        this._renderedPages = [];
+
+        if (this._pages.length <= 0) {
+            return undefined;
+        }
+
+        const cardLevelContainer: HTMLElement = document.createElement("div");
+
+        const containerForAdorners: HTMLElement = document.createElement("div");
+        containerForAdorners.className = this.hostConfig.makeCssClassName("ac-carousel-container");
+        cardLevelContainer.appendChild(containerForAdorners);
+
+        const swiperContainer: HTMLElement = document.createElement("div");
+        swiperContainer.className = this.hostConfig.makeCssClassName("swiper", "ac-carousel");
+
+        const swiperWrapper: HTMLElement = document.createElement("div");
+        swiperWrapper.className = this.hostConfig.makeCssClassName("swiper-wrapper", "ac-carousel-card-container");
```

Comment:
Since our official name is carousel, can we rename all swiper to carousel in code?